### ISSUE TITLE
Bug: useTheme() reports incorrect type since v1.52.7

### ISSUE THREAD

**Issue by nderscore** [tamagui/tamagui#1563]
### Current Behavior

Since v1.52.7, the return type of `useTheme()` returns a different shape for theme variables that doesn't reflect the actual shape.

`useThemeResult.{themeKey}.val` is now reported by the types as a nested variable:

![image](https://github.com/tamagui/tamagui/assets/2250252/44652614-023c-4d16-965e-966fd9a893b6)

But the changes reflected by these types don't seem to be real?

The expression `useThemeResult.{themeKey}.val` still resolves to the raw value as it did before in previous versions.

### Expected Behavior

This was the previous shape of the types:

![image](https://github.com/tamagui/tamagui/assets/2250252/2516c796-ab54-4c19-9523-85d335db2c4d)

### Tamagui Version

```markdown
1.52.7 - 1.53.1
```

### Reproduction

```markdown
This happens when creating themes using colors from tokens:


import { createTheme, createTokens } from '@tamagui/core'

const tokens = createTokens({ /* ... */ })

const lightTheme = createTheme({
  color: tokens.color.foo123,
  // ...
})
```

**Comment by natew** [comment]
fixed v1.53.2

**Comment by nderscore** [comment]
@natew here's a reproduction in 1.53.2: 

https://github.com/nderscore/tamagui-usetheme-val

**Comment by nderscore** [comment]
The fix in v1.54.1 seems to be working 🙏 Thanks Nate!
